coalib/core/DependencyTracker.py

```diff
@@ -0,0 +1,118 @@
+class DependencyTracker:
+    """
+    A ``DependencyTracker`` allows to register and manage dependencies between
+    objects.
+
+    This class uses a directed graph to track relations.
+
+    Add a dependency relation between two objects:
+
+    >>> object1 = object()
+    >>> object2 = object()
```

Comment:
`object` shall demonstrate that you can use arbitrary objects + a new object is unique